File: count-prefix-and-suffix-pairs-i.ts
{
    function countPrefixSuffixPairs(words: string[]): number {
        let c = 0;

        for (let i = 0; i < words.length - 1; i++) {
            const first_word = words[i];

            for (let j = i + 1; j < words.length; j++) {
                const second_word = words[j];

                if(second_word.startsWith(first_word) && second_word.endsWith(first_word))
                    c += 1;
            }
            
        }

        return c
    };  

    console.log(countPrefixSuffixPairs(["a","aba","ababa","aa"]))
}

File: count-prefix-and-suffix-pairs-i.rs
pub fn count_prefix_suffix_pairs(words: Vec<String>) -> i32 {
    let mut c = 0;

    for i in 0..words.len() - 1 {
        let first_word = &words[i];

        for j in i + 1..words.len() {
            let second_word = &words[j];

            if second_word.starts_with(first_word) && second_word.ends_with(first_word) {
                c += 1;
            }
        }

    }

    return c;
}